lib/src/webcrypto/webcrypto.ecdh.dart

```diff
@@ -14,43 +14,250 @@
+  /// * `"crv": "P-384"`, use [EllipticCurve.p384], and,
+  /// * `"crv": "P-521"`, use [EllipticCurve.p521].
+  /// 
+  /// **Example**
+  /// ```dart
+  /// import 'package:webcrypto/webcrypto.dart';
+  /// 
+  /// // JSON Web Key as a string containing JSON.
+  /// final jwk = {
+  ///   'kty': 'EC',
+  ///   'crv': 'P-256',
+  ///   'x': 'kgR_PqO07L8sZOBbw6rvv7O_f7clqDeiE3WnMkb5EoI',
+  ///   'y': 'djI-XqCqSyO9GFk_QT_stROMCAROIvU8KOORBgQUemE',
+  ///   'd': '5aPFSt0UFVXYGu-ZKyC9FQIUOAMmnjzdIwkxCMe3Iok',
+  /// };
+  /// 
+  /// Future<void> main() async {
+  ///   // Import secret key from decoded JSON.
+  ///   final jsonWebKey = await EcdhPrivateKey.importJsonWebKey(
+  ///     jwk,
+  ///     EllipticCurve.p256,
+  ///   );
+  /// 
+  ///   // Export the key (print it in same format as it was given).
+  ///   final exportedJsonWebKey = await jsonWebKey.exportJsonWebKey();
+  ///   print(exportedJsonWebKey);
+  /// }
+  /// ```
+  /// 
+  /// [1]: https://www.rfc-editor.org/rfc/rfc7518.html#section-6.2
   static Future<EcdhPrivateKey> importJsonWebKey(
     Map<String, dynamic> jwk,
     EllipticCurve curve,
   ) {
     return impl.ecdhPrivateKey_importJsonWebKey(jwk, curve);
   }
 
+  /// Generate a new [EcdhPrivateKey] and [EcdhPublicKey] pair.
+  ///
+  /// The [curve] parameter specifies the curve to use for the key pair.
+  /// 
+  /// Use [EllipticCurve.p256] for the P-256 curve.
+  /// Use [EllipticCurve.p384] for the P-384 curve.
+  /// Use [EllipticCurve.p521] for the P-521 curve.
+  /// 
+  /// **Example**
+  /// ```dart
+  /// import 'package:webcrypto/webcrypto.dart';
+  /// 
+  /// Future<void> main() async {
+  ///   // Generate a new key pair using the P-256 curve.
+  ///   final keyPair = await EcdhPrivateKey.generateKey(EllipticCurve.p256);
+  /// 
+  ///   // Export the private key.
+  ///   final exportedPrivateKey = await keyPair.privateKey.exportJsonWebKey();
+  ///   print(exportedPrivateKey);
+  /// 
+  ///   // Export the public key.
+  ///   final exportedPublicKey = await keyPair.publicKey.exportJsonWebKey();
+  ///   print(exportedPublicKey);
+  /// }
+  /// ```
+  /// 
   static Future<KeyPair<EcdhPrivateKey, EcdhPublicKey>> generateKey(
     EllipticCurve curve,
   ) {
     return impl.ecdhPrivateKey_generateKey(curve);
   }
 
+  /// Derive an array of bits from the [EcdhPrivateKey].
+  ///
+  /// The [length] parameter specifies the number of bits to derive and 
+  /// should be multiples of 8.
+  /// The [publicKey] parameter is the [EcdhPublicKey].
+  /// 
+  /// **Example**
+  /// ```dart
+  /// import 'package:webcrypto/webcrypto.dart';
+  /// 
+  /// Future<void> main() async {
+  ///   // Generate a new key pair using the P-256 curve.
+  ///   final keyPair = await EcdhPrivateKey.generateKey(EllipticCurve.p256);
+  /// 
+  ///   // Derive 256 bits from the private key using the public key.
+  ///   final derivedBits = await keyPair.privateKey.deriveBits(256, keyPair.publicKey);
```

Comment:
When is usage such as this legitimate?

Is it wrong to use the same public / private key pair together?
I don't know, but I thought the idea was that you generated two key-pairs, exchanged public keys and then generated the same shared secret.

Using a single key-pair may work, but what is the point? If you're sending the private key over a secure channel, then you have a secure channel and might as well send a randomly generated shared secret instead.

At best this is an example of a fairly complex and expensive RNG, at worst this is a flared RNG.
I don't have to expertise to say whether this is a reasonable way to generate random numbers. But I don't think this is how ECDH is typically intended to be used.